### PR TITLE
FIX #1398 detect epiphany

### LINKED ISSUE #1398: Epiphany browser detect as Safari
Tested on Epiphany 3.14.1 (WebKit 2.6.2)

```
Browser Name: Safari

Browser Version: 6

Browser UA String: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/601.1 (KHTML, like Gecko) Version/8.0 Safari/601.1 Debian/unstable (3.14.1-1) Epiphany/3.14.1
```

### PR DESCRIPTION
**Pull request by wanton1950**
Fix for #1398 

Detects epiphany browser by looking for Safari and then looking for Epiphany.
Added browser name code: `EP`
Browser version is obtained via believing the UserAgent string.

I can see hat you mean by needing to refactor the browser detection - more on that in an appropriate location -- #1392 

Unrelated whitespace changes in files, mostly `browser.js`, indentation consistency mostly. 

Tested using a Epiphany on a Raspberry Pi. 
On a related note: Chromium on the Pi is not detected correctly.

### CODE REVIEW

File: core/main/client/browser.js [hunk 1]
```diff
@@ -646,6 +646,16 @@ beef.browser = {
         return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
     },
 
+    /**
+     * Return true if Epiphany
+     * @example: beef.browser.isEpi()
+     */
+    isEpi: function () {
+        // based on webkit -- therefore should look a bit like Safari
+        return this.isS() && window.navigator.userAgent.match(/Epiphany\//) != null;
```
Comment (bcoles) [inline core/main/client/browser.js]: This shouldn't rely on `isS()`

File: core/main/client/browser.js [hunk 2]
```diff
@@ -2305,6 +2317,18 @@ beef.browser = {
         }
         ;   // Microsoft Edge
 
+        if (this.isEpi()) {
+            // beleive the UserAgent string - until whenever
```
Comment (bcoles) [inline core/main/client/browser.js]: `believe`

File: core/main/client/browser.js [hunk 3]
```diff
@@ -2305,6 +2317,18 @@ beef.browser = {
...
+            var epiphanyRe = /Epiphany\/(\d+)/;
+            var versionDetails = epiphanyRe.exec( beef.browser.getBrowserReportedName());
+            if (versionDetails.length > 1) {
+                return versionDetails[1];
+            } else {
+                return "UNKNOWN";
+            }
+        }
+        ;                       // Epihany
```
Comment (bcoles) [inline core/main/client/browser.js]: `Epiphany`

File: core/main/client/browser.js [hunk 4]
```diff
@@ -2305,6 +2317,18 @@ beef.browser = {
...
         ;   // Microsoft Edge
 
+        if (this.isEpi()) {
+            // beleive the UserAgent string - until whenever
+            var epiphanyRe = /Epiphany\/(\d+)/;
+            var versionDetails = epiphanyRe.exec( beef.browser.getBrowserReportedName());
+            if (versionDetails.length > 1) {
+                return versionDetails[1];
+            } else {
+                return "UNKNOWN";
```
Comment (bcoles) [inline core/main/client/browser.js]: The `is*()` methods should return a Boolean.

**Comment by bcoles** [comment]
Whitespace cleanup is great, but please try to keep unrelated refactoring out of PRs.

File: core/main/client/browser.js [hunk 5]
```diff
@@ -2379,6 +2403,10 @@ beef.browser = {
             return 'O'
         }
         ;		// Opera any version
+        if (this.isEpi()) {
+            return 'EP'
+        }
+        ;			// Ephipany any version
```
Comment (bcoles) [inline core/main/client/browser.js]: `Epiphany`

**Comment by wanton1950** [comment]
Patch updated.
Corrected speeling mistakes.
Removed IsS() from detection - relying on the UserAgent string.
Additional comments.

File: core/main/client/browser.js [hunk 6]
```diff
@@ -2305,6 +2317,18 @@ beef.browser = {
...
 
+        if (this.isEpi()) {
+            // believe the UserAgent string for version info - until whenever
+            var epiphanyRe = /Epiphany\/(\d+)/;
+            var versionDetails = epiphanyRe.exec( beef.browser.getBrowserReportedName());
+            if (versionDetails.length > 1) {
+                return versionDetails[1];
+            } else {
+                return "UNKNOWN"; // returns from here or it may take Safari version details
+            }
```
Comment (bcoles) [inline core/main/client/browser.js]:
```javascript
            if (versionDetails.length > 1) {
                return true;
            } else {
                return false;
            }
```

File: core/main/client/browser.js [hunk 7]
```diff
@@ -646,6 +646,16 @@ beef.browser = {
         return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
     },
 
+    /**
+     * Return true if Epiphany
+     * @example: beef.browser.isEpi()
+     */
+    isEpi: function () {
+        // based on webkit -- therefore should look a bit like Safari
+        return window.navigator.userAgent.match(/Epiphany\//) != null;
```
Comment (bcoles) [inline core/main/client/browser.js]: This should be based on more than just the user agent. What part of `isS()` was matching?

File: core/main/client/browser.js (hunk 6 above)
Comment (wanton1950) [reply core/main/client/browser.js]: This function returns a string -- returning a boolean will break things

Top of the function:

Line: 1677
`    /**
     * Returns the type of browser being used.
     * @return: {String} User agent software and version.
     *
     * @example: beef.browser.getBrowserVersion()
     */
    getBrowserVersion: function () {
`

Bottom of the function:  line 2376.

`
        return 'UNKNOWN';				// Unknown UA
    },
`

File: core/main/client/browser.js [hunk 8]
```diff
@@ -646,6 +646,27 @@ beef.browser = {
         return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
     },
 
+    /**
+     * Returns true if Webkit based
+     */
+    isWebKitBased: function () {
+        // Currently, [2018-01-08], safari checks isS* provide evidence of webkit
+        // to avoid duplicating code and hence bugs, use the existing checks for Safari.
+        return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
```
Comment (bcoles) [inline core/main/client/browser.js]: `isWebKitBased` should not depend on `isS*()` even if it means code duplication.

**Comment by wanton1950** [comment]
Made the isWebKit more explicit -- resisted the urge to change all the isS\d functions  ^_^
Although if you wanted that -- this would be good time to do it.

**Comment by bcoles** [comment]
Thanks

**Pull request merged**